Adds local and CI tests

randyhbh opened the pull request:
Adds local browser-based unit tests and CI/CD workflow for the Chrome extension.

- Enables automated testing using JSDOM and GitHub Actions.
- Improves code quality and reliability through decoupled logic and comprehensive unit tests.

Commit:
feat: Initialize Node.js project and .gitignore

Commit:
test: Add browser-based unit tests for content script

Commit:
docs: Add CLAUDE.md for AI project guidance

Commit:
ci: Set up headless testing with JSDOM and GitHub Actions

Commit:
refactor: Decouple content.js logic for testability

Commit:
test: Verify position styles in getButtonStyles

Commit:
chore: Update test results timestamp

Commit:
test: Add edge case tests for parseRawGitHubUrl

Commit:
refactor: Introduce class-based test runner

Commit:
feat: Enhance button mocking and test coverage for creation/styling

jadolg commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,34 @@
+
+on:
+  push:
+    branches: [ main, develop ]
```
No develop branch needed here.

jadolg commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,34 @@
+      with:
+        name: test-results
+        path: test-results.json
+        retention-days: 7
```
Please add an empty line at the end of the file.

jadolg commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,34 @@
+    - name: Run tests
+      run: npm test
+      
+    - name: Upload test results
```
This step should not be needed

jadolg commented on .gitignore:
```diff
@@ -0,0 +1,83 @@
+### JetBrains template
```
This file is too general. Better to just ignore `node_modules` and `.idea` in this case. The rest is never gonna be there.

jadolg commented on package.json:
```diff
@@ -0,0 +1,14 @@
+  "keywords": ["chrome-extension", "github", "url-converter"],
+  "author": "",
+  "license": "MIT"
+}
```
please add new line

jadolg commented on test-results.json:
```diff
@@ -0,0 +1,61 @@
+{
```
This file should not be versioned since it's just the tests results

jadolg commented on CLAUDE.md:
```diff
@@ -0,0 +1,71 @@
+# CLAUDE.md
```
Do we need to version this?

Commit:
CI: Limit push trigger for test workflow to main branch

Commit:
Chore: Apply consistent formatting and ensure file newlines

Commit:
Chore: Refine .gitignore for project-specific exclusions

Commit:
chore: Remove AI guidance file and generated test results

Commit:
feat: Add EditorConfig for coding styles & better IDE integration

Commit:
ci: Add style check to GitHub Actions workflow

Commit:
feat: Implement script for end-of-file newline check

Current situation: this project is intentionally lightweight (only jsdom dependency) and the custom script works well for CI enforcement.

Commit:
fix: Ensure all files end with a newline
